### PR TITLE
[KernelGen][Nvidia] Add rot90 operator with Triton kernel

### PR DESCRIPTION
**Pull request by XDYuanzhuLee**
## Summary
Adds a Triton kernel for `rot90`. Triton kernel implementation for `rot90`.

## Testing
- Validated against reference on device via `to_reference(inp, True)`
- Tested on: Nvidia, Tianshu, Muxi, Ascend, Hygon

## Performance
Test command: `pytest benchmark/test_rot90.py --level core` (NVIDIA H20)

| dtype | Size | k | dims | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|------|---|------|--------------------|--------------------|---------|
| float16 | [64, 64] | 1 | [0, 1] | 0.011264 | 0.007168 | 1.571 |
| float16 | [128, 128] | 1 | [0, 1] | 0.010240 | 0.008192 | 1.250 |
| float16 | [256, 256] | 1 | [0, 1] | 0.010240 | 0.009216 | 1.111 |
| float16 | [512, 512] | 1 | [0, 1] | 0.010240 | 0.010240 | 1.000 |
| float16 | [1024, 1024] | 1 | [0, 1] | 0.014336 | 0.017408 | 0.824 |
| float16 | [2048, 2048] | 1 | [0, 1] | 0.026624 | 0.046080 | 0.578 |
| float16 | [100, 200] | 1 | [0, 1] | 0.010240 | 0.008192 | 1.250 |
| float16 | [200, 400] | 1 | [0, 1] | 0.010240 | 0.008192 | 1.250 |
| float16 | [400, 800] | 1 | [0, 1] | 0.011264 | 0.011264 | 1.000 |
| float32 | [64, 64] | 1 | [0, 1] | 0.009216 | 0.008192 | 1.125 |
| float32 | [128, 128] | 1 | [0, 1] | 0.010240 | 0.008192 | 1.250 |
| float32 | [256, 256] | 1 | [0, 1] | 0.010240 | 0.008192 | 1.250 |
| float32 | [512, 512] | 1 | [0, 1] | 0.011264 | 0.011264 | 1.000 |
| float32 | [1024, 1024] | 1 | [0, 1] | 0.016384 | 0.020480 | 0.800 |
| float32 | [2048, 2048] | 1 | [0, 1] | 0.036864 | 0.058368 | 0.632 |
| float32 | [100, 200] | 1 | [0, 1] | 0.010240 | 0.008192 | 1.250 |
| float32 | [200, 400] | 1 | [0, 1] | 0.010240 | 0.009216 | 1.111 |
| float32 | [400, 800] | 1 | [0, 1] | 0.012288 | 0.012288 | 1.000 |
| bfloat16 | [64, 64] | 1 | [0, 1] | 0.009216 | 0.007168 | 1.286 |
| bfloat16 | [128, 128] | 1 | [0, 1] | 0.009216 | 0.008192 | 1.125 |
| bfloat16 | [256, 256] | 1 | [0, 1] | 0.010240 | 0.009216 | 1.111 |
| bfloat16 | [512, 512] | 1 | [0, 1] | 0.011264 | 0.010240 | 1.100 |
| bfloat16 | [1024, 1024] | 1 | [0, 1] | 0.015360 | 0.017408 | 0.882 |
| bfloat16 | [2048, 2048] | 1 | [0, 1] | 0.027648 | 0.045056 | 0.614 |
| bfloat16 | [100, 200] | 1 | [0, 1] | 0.010240 | 0.008192 | 1.250 |
| bfloat16 | [200, 400] | 1 | [0, 1] | 0.010240 | 0.009216 | 1.111 |
| bfloat16 | [400, 800] | 1 | [0, 1] | 0.011264 | 0.011264 | 1.000 |
| **Arithmetic Mean** | — | — | — | — | — | **1.075** |

## Multi-backend Testing
| Backend | Accuracy Test | Speedup (mean) | Notes |
|---|---|---|---|
| Nvidia (H20) | PASS (27 cases) | 1.075 | Primary |
| Tianshu | PASS | 1.242 | — |
| Muxi | PASS | 1.151 | — |
| Ascend | PASS | — | — |
| Hygon | FAIL | 0.911 | error: operand #0 does not dominate this use |

## Files Changed
- `src/flag_gems/ops/rot90.py`: Triton kernel implementation
- `tests/test_rot90.py`: Accuracy test
- `benchmark/test_rot90.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: Math, stage: alpha 5.1)

### CODE REVIEW

File: src/flag_gems/ops/rot90.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+import triton
+import triton.language as tl
+
+from flag_gems.runtime import torch_device_fn
+
+logger = logging.getLogger(__name__)
+
+
+# Autotuning configurations
+@triton.autotune(
```
Comment (bin913) [inline src/flag_gems/ops/rot90.py]: **Suggestion**: Large inline autotune config (4 configs hardcoded in kernel)

**Recommended fix:**
Move autotune configs to a config file instead of hardcoding inline.

<sub>Rule: `KERNEL_AUTOTUNE_CONFIG`</sub>

**Review by bin913 — CHANGES_REQUESTED** [review]
## FlagGems Automated Review

**Operator:** `rot90`
**Verdict:** REQUEST_CHANGES

### Summary
- Errors: 1
- Warnings: 1
- Suggestions: 0

### Issues (no specific line)
- **[ERROR]** `GIT_COAUTHOR`: Latest commit contains Co-Authored-By trailer — *Fix: Remove Co-Authored-By from the latest commit to pass CLA CI.*

File: src/flag_gems/ops/rot90.py (hunk 1 above)
Comment (bin913) [inline src/flag_gems/ops/rot90.py]: **Suggestion**: Large inline autotune config (4 configs hardcoded in kernel)

**Recommended fix:**
Move autotune configs to a config file instead of hardcoding inline.

<sub>Rule: `KERNEL_AUTOTUNE_CONFIG`</sub>

**Review by bin913 — CHANGES_REQUESTED** [review]
## FlagGems Automated Review

**Operator:** `rot90`
**Verdict:** REQUEST_CHANGES

### Summary
- Errors: 1
- Warnings: 1
- Suggestions: 0

### Issues (no specific line)
- **[ERROR]** `GIT_COAUTHOR`: Latest commit contains Co-Authored-By trailer — *Fix: Remove Co-Authored-By from the latest commit to pass CLA CI.*

File: src/flag_gems/ops/rot90.py (hunk 1 above)
Comment (XDYuanzhuLee) [reply src/flag_gems/ops/rot90.py]: Two fixes applied:

1. **Autotune configs**: Moved the 4 rot90 autotune configurations from the kernel decorator to `src/flag_gems/runtime/backend/_nvidia/tune_configs.yaml` as a new `rot90` entry. The kernel now loads configs via `runtime.get_tuned_config("rot90")`.

2. **Co-Authored-By trailer**: The new commit (4b915642) has no forbidden trailers.

Validation: `python -m py_compile src/flag_gems/ops/rot90.py` PASS; `pre-commit run --files src/flag_gems/ops/rot90.py src/flag_gems/runtime/backend/_nvidia/tune_configs.yaml` PASS (YAML check, isort, black, flake8).

The duplicate autotune comment at #discussion_r3334393842 and duplicate GIT_COAUTHOR review at #pullrequestreview-4401475420 report the same issues — both are addressed.

Comment (XDYuanzhuLee) [reply src/flag_gems/ops/rot90.py]: Two fixes applied:

1. **Autotune configs**: Moved the 4 rot90 autotune configurations from the kernel decorator to `src/flag_gems/runtime/backend/_nvidia/tune_configs.yaml` as a new `rot90` entry. The kernel now loads configs via `runtime.get_tuned_config("rot90")`.

2. **Co-Authored-By trailer**: The new commit (4b915642) has no forbidden trailers.

Validation: `python -m py_compile src/flag_gems/ops/rot90.py` PASS; `pre-commit run --files src/flag_gems/ops/rot90.py src/flag_gems/runtime/backend/_nvidia/tune_configs.yaml` PASS (YAML check, isort, black, flake8).

The duplicate autotune comment at #discussion_r3334393842 and duplicate GIT_COAUTHOR review at #pullrequestreview-4401475420 report the same issues — both are addressed.

**Review by bin913 — APPROVED** [review]
